src/treeinterp/BTFactory.cpp

```diff
@@ -19,6 +28,14 @@ void BTFactory::makeTrees() {
 
     std::cout << "Re-Make Trees From Json" << std::endl;
 
+    /*
+     * Here we store the C++ trees in a map, key = treename, val = cpp tree.
+     * In order to do this in a cleaner way, maybe build trees automatically by going through directory
+     */
+    auto Off = new bt::OffensiveStrategy();
```

Comment:
Memory leak.

Change to 

auto Off = bt::OffensiveStrategy();

codeTrees[...] = Off.createOffensiveStrategy();